### ISSUE TITLE
headless tests on travis don't antialias fills

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#1555]
Actual:
![image](https://cloud.githubusercontent.com/assets/98601/7624280/20dc7742-f99b-11e4-908a-b8cfc99845b9.png)

Expected:
![image](https://cloud.githubusercontent.com/assets/98601/7624277/1838bdbc-f99b-11e4-83a5-152d48947067.png)

**Comment by jfirebaugh** [comment]
This appears to be an issue with `glLineWidth` / `GL_LINES`. If I switch this test case to just draw the outline, I get the following result locally:

![image](https://cloud.githubusercontent.com/assets/98601/7695601/b3d9e976-fda6-11e4-9681-78db0237a7f6.png)

And this result on linux:

![image](https://cloud.githubusercontent.com/assets/98601/7695602/bcfd9e08-fda6-11e4-9f4d-a587a87b7cef.png)

Fiddling with the line width on linux doesn't seem to produce any better results. E.g., here line width = 10:

![image](https://cloud.githubusercontent.com/assets/98601/7695617/e02d314a-fda6-11e4-9647-b4366963cbe3.png)

`glGet` reports `GL_LINE_WIDTH_RANGE` = 1 - 10 and `GL_LINE_WIDTH_GRANULARITY` = 0.1.

@kkaefer Any ideas?

**Comment by jfirebaugh** [comment]
Here's a hint: if I remove the multiplication by `alpha` [here](https://github.com/mapbox/mapbox-gl-native/blob/abafb52f37beb5659efc2105ccd1568e1f754898/src/mbgl/shader/outline.fragment.glsl#L8-8), it looks like this:

![image](https://cloud.githubusercontent.com/assets/98601/7695671/d815c8ea-fda7-11e4-9fb0-10034af34c9b.png)

It looks like perhaps every other segment is simply not being drawn.

**Comment by kkaefer** [comment]
I can reproduce this behavior on Ubuntu 15.05, but my output looks different:

![actual](https://cloud.githubusercontent.com/assets/52399/7723560/feb3ec82-fee9-11e4-9a03-8ddaccd1f29e.png)

**Comment by kkaefer** [comment]
Playing with this a bit further, I'm seeing only mostly horizontal lines:

![image](https://cloud.githubusercontent.com/assets/52399/7731190/120d50ea-ff1f-11e4-8a99-1c99de1796ad.png)

**Comment by kkaefer** [comment]
Looks like our shader fails for lines that are more than 45 degrees off horizontally:

| Ubuntu | OS X |
| --- | --- |
| ![ubuntu](https://cloud.githubusercontent.com/assets/52399/7745267/2838e9d4-ffaa-11e4-88d3-e3ed8658598c.png) | ![osx](https://cloud.githubusercontent.com/assets/52399/7745270/2c4673ca-ffaa-11e4-92f5-3155acaefb29.png) |

**Comment by kkaefer** [comment]
Further investigation shows that line rasterization seems fundamentally different on mesa compared to my (OS X) implementation:

| Ubuntu | OS X |
| --- | --- |
| ![ubuntu](https://cloud.githubusercontent.com/assets/52399/7752958/bf747966-ffe2-11e4-9cc9-7c10afb58046.png) | ![osx](https://cloud.githubusercontent.com/assets/52399/7752963/c4d2b2ec-ffe2-11e4-9e58-75e4d127906d.png) |

![diff](https://cloud.githubusercontent.com/assets/52399/7752981/d8b4ed2a-ffe2-11e4-94a0-9a020601dbf6.png)

I couldn't figure out a correct offsetting that gives (nearly) identical results on Mesa and OS X. My hunch is that Mesa has a different default layout qualifier than OS X, since offsetting the glFragCoord by half a pixel in both directions yields a better result (see above). Yet, the lines don't quite match in terms of intensity.

**Comment by kkaefer** [comment]
Not sure if this is related: https://www.opengl.org/sdk/docs/man/html/gl_FragCoord.xhtml specifies the ability to change the pixel center to integers with `pixel_center_integer`

**Comment by kkaefer** [comment]
for reference, the above images were created with

```
OpenGL vendor string: VMware, Inc.
OpenGL renderer string: Gallium 0.4 on SVGA3D; build: RELEASE;  
OpenGL version string: 2.1 Mesa 10.5.2
```

**Comment by jfirebaugh** [comment]
Let's check if this affects headless rendering. If so, we definitely need to find a fix or switch to triangulated outlines.

**Comment by jfirebaugh** [comment]
This is apparently no longer an issue in gl-js, now that it's using headless-gl 2.0. It's still a problem in gl-native.

**Comment by brunoabinader** [comment]
We could alleviate this issue by either 1) switching node bot to macOS on Travis or 2) forcefully disabling antialias in both -js and -native and update expectations.

**Comment by mikemorris** [comment]
Should have noted this earlier, but FWIW Travis CI is rendering with Mesa's [software rasterizer](https://github.com/mapbox/mason/blob/98feac14b1ae298322ee30291105b40ba29dac0b/scripts/mesa/11.2.2/script.sh#L23) (with [llvmpipe](http://www.mesa3d.org/llvmpipe.html)), not a hardware GPU, but given @kkaefer's comment in https://github.com/mapbox/mapbox-gl-native/issues/1555#issuecomment-104339130 this could just be an issue with Mesa that isn't specific to the software rasterizer.

**Comment by brunoabinader** [comment]
Fixed in #6899.

**Comment by kkaefer** [comment]
🎉
